utils/obsproc/IcecAmsr2Ioda.h

```diff
@@ -0,0 +1,117 @@
+      : NetCDFToIodaConverter(fullConfig) {
+      variable_ = "seaIceFraction";
+    }
+
+    // Read netcdf file and populate iodaVars
+    gdasapp::IodaVars providerToIodaVars(const std::string fileName) final {
+      oops::Log::info() << "Processing files provided by the AMSR2" << std::endl;
+
+      // Open the NetCDF file in read-only mode
+      netCDF::NcFile ncFile(fileName, netCDF::NcFile::read);
+
+      // Get the number of obs in the file
+      int dimxSize = ncFile.getDim("Number_of_X_Dimension").getSize();
+      int dimySize = ncFile.getDim("Number_of_Y_Dimension").getSize();
+      int dimTimeSize = ncFile.getDim("Time_Dimension").getSize();
+      int nobs = dimxSize * dimySize;
+      int ntimes = dimxSize * dimySize * dimTimeSize;
+
+      // Set the int metadata names
+      std::vector<std::string> intMetadataNames = {};
+      // std::vector<std::string> intMetadataNames = {"pass", "cycle", "mission"};
+
+      // Set the float metadata name
+      std::vector<std::string> floatMetadataNames = {};
+      // std::vector<std::string> floatMetadataNames = {"mdt"};
+      // Create instance of iodaVars object
+      gdasapp::IodaVars iodaVars(nobs, floatMetadataNames, intMetadataNames);
+
+      oops::Log::debug() << "--- iodaVars.location: " << iodaVars.location << std::endl;
+
+      // Read non-optional metadata: datetime, longitude and latitude
+      std::vector<float> oneDimLatVal(iodaVars.location);
+      ncFile.getVar("Latitude").getVar(oneDimLatVal.data());
+
+      std::vector<float> oneDimLonVal(iodaVars.location);
+      ncFile.getVar("Longitude").getVar(oneDimLonVal.data());
+
+      // Read optional integer metadata "Flags"
+      std::vector<int64_t> oneDimFlagsVal(iodaVars.location);
+      ncFile.getVar("Flags").getVar(oneDimFlagsVal.data());
+
+      // Get Ice_Concentration obs values and attributes
+      std::vector<int> oneDimObsVal(iodaVars.location);
+      ncFile.getVar("NASA_Team_2_Ice_Concentration").getVar(oneDimObsVal.data());
+
+      // Read and process the dateTime
+      std::vector<int> oneTmpdateTimeVal(ntimes);
+      ncFile.getVar("Scan_Time").getVar(oneTmpdateTimeVal.data());
+      iodaVars.referenceDate = "seconds since 1970-01-01T00:00:00Z";
+
+      std::tm timeinfo = {};
+      for (int i = 0; i < ntimes; i += dimTimeSize) {
+        for (int j = 0; j < dimTimeSize && i + j < ntimes; j++) {
+        }
+        timeinfo.tm_year = oneTmpdateTimeVal[i] - 1900;  // Year since 1900
+        timeinfo.tm_mon = oneTmpdateTimeVal[i + 1] - 1;  // 0-based; 8 represents Sep
+        timeinfo.tm_mday = oneTmpdateTimeVal[i + 2];
+        timeinfo.tm_hour = oneTmpdateTimeVal[i + 3];
+        timeinfo.tm_min = oneTmpdateTimeVal[i + 4];
+        timeinfo.tm_sec = oneTmpdateTimeVal[i + 5];
+
+        // Calculate and store the seconds since the Unix epoch
+        time_t epochtime = std::mktime(&timeinfo);
+        iodaVars.datetime(i/6) = static_cast<int64_t>(epochtime);
+      }
```

Comment:
with a bit of grepping, I found utilities for datetime manipulation under ```ioda/src/core/IodaUtils.h```. Were you aware of these? If so, any reasons why not use what's in IodaUtils?